src/utm.ml

```diff
@@ -1 +1,38 @@
-let name () = Printf.printf "UTM\n"
+module CharSet = Set.Make(Char)
+
+type state = int
+type direction = L | R
+type echar = Blank | Char of char
+type tape = echar list * echar * echar list
+type status = Running | Accepted | Rejected
+
+type t = CharSet.t                                     (* Alphabet *)
+       * string array                                  (* State Mapping *)
+       * state                                         (* Current State *)
+       * (state * echar) -> (direction * echar * state)(* Transition Function *)
+       * state                                         (* Accept State *)
+       * state                                         (* Reject State *)
+       * tape                                          (* Infinite Tape *)
+
+
+(* Possible optimization - Prune Blanks from ends of list *)
+let moveHead = function
+	| (([],c,r), L) -> ([],Blank,c::r)
+	| ((c'::l,c,r), L) -> (l,c',c::r)
+	| ((l,c,[]), R) -> (c::l,Blank,[])
+	| ((l,c,c'::r), R) -> (c::l,c',r)
+
+let writeChar (l,_,r) c' = (l,c',r)
+
+(* step: t -> t * status
+ * step t evaluates one step of the turing machine specified by t. It outputs 
+ * the resulting turing machine and status of the machine after the step.
+ *
+ * NOTE: Currently doesn't handle case where initial state is accept/reject
+ *)
+let step = function
+	| (alpha,labels,s,delta,a,r,((_,c,_) as t)) ->
+		let (dir,c',s'):(direction * echar * state) = delta (s,c) in
+		let t' = moveHead ((writeChar t c'),dir) in
+		let res = if s'= a then Accepted else if s'=r then Rejected else Running in
```

Comment:
The status isn't actually part of the TM. It's just another thing that's returned by the step function. So right now the function type is like `ocaml Utm.t -> Utm.t * Utm.status`. The purpose is that the driver will probably eventually want that information since it will be advancing the TM step by step. Maybe instead we can do the check at the beginning and return the status and the TM unchanged if it is already accepting or rejecting.